vlog4j-parser/src/main/java/org/semanticweb/vlog4j/parser/javacc/JavaCCParserBase.java

```diff
@@ -222,6 +235,24 @@ void resetVariableSets() {
 		this.headUniVars.clear();
 	}
 
+	/**
+	 * Convert a throwable into a ParseException.
+	 *
+	 * @param message The error message.
+	 * @param cause   The {@link Throwable} that caused this exception.
+	 *
+	 * @return A {@link ParseException} with appropriate cause and message.
+	 */
+	protected ParseException makeParseExceptionWithCause(String message, Throwable cause) {
+		ParseException parseException = new ParseException(message);
+		parseException.initCause(cause);
+		return parseException;
+	}
+
+	protected ParseException makeParseExceptionWithCause(Throwable cause) {
+		return makeParseExceptionWithCause(cause.getMessage(), cause);
```

Comment:
This is not necessary, as the cause exception message is in the exception trace if the cause is propagated.


```
Exception in thread "main" .SomeException: someExceptionMessage
	at ...
Caused by: CauseException: causeExceptionMessage
```